backend/infrahub/core/schema/definitions/core/__init__.py

```diff
@@ -1336,71 +1098,8 @@
                 },
             ],
         },
-        {
-            "name": "Repository",
-            "namespace": "Core",
-            "description": "A Git Repository integrated with Infrahub",
-            "include_in_menu": False,
-            "icon": "mdi:source-repository",
-            "label": "Repository",
-            "default_filter": "name__value",
-            "order_by": ["name__value"],
-            "display_labels": ["name__value"],
-            "generate_profile": False,
-            "branch": BranchSupportType.AGNOSTIC.value,
-            "inherit_from": [
-                InfrahubKind.LINEAGEOWNER,
-                InfrahubKind.LINEAGESOURCE,
-                InfrahubKind.GENERICREPOSITORY,
-                InfrahubKind.TASKTARGET,
-            ],
-            "documentation": "/topics/repository",
-            "attributes": [
-                {"name": "default_branch", "kind": "Text", "default_value": "main", "order_weight": 6000},
-                {
-                    "name": "commit",
-                    "kind": "Text",
-                    "optional": True,
-                    "branch": BranchSupportType.LOCAL.value,
-                    "order_weight": 7000,
-                },
-            ],
-        },
-        {
-            "name": "ReadOnlyRepository",
-            "namespace": "Core",
-            "description": "A Git Repository integrated with Infrahub, Git-side will not be updated",
-            "include_in_menu": False,
-            "label": "Read-Only Repository",
-            "default_filter": "name__value",
-            "order_by": ["name__value"],
-            "display_labels": ["name__value"],
-            "generate_profile": False,
-            "branch": BranchSupportType.AGNOSTIC.value,
-            "inherit_from": [
-                InfrahubKind.LINEAGEOWNER,
-                InfrahubKind.LINEAGESOURCE,
-                InfrahubKind.GENERICREPOSITORY,
-                InfrahubKind.TASKTARGET,
-            ],
-            "documentation": "/topics/repository",
-            "attributes": [
-                {
-                    "name": "ref",
-                    "kind": "Text",
-                    "default_value": "main",
-                    "branch": BranchSupportType.AWARE.value,
-                    "order_weight": 6000,
-                },
-                {
-                    "name": "commit",
-                    "kind": "Text",
-                    "optional": True,
-                    "branch": BranchSupportType.AWARE.value,
-                    "order_weight": 7000,
-                },
-            ],
-        },
+        core_repository.to_dict(),
```

Comment:
Might be cleaner to have a temporary dictionary that allows for both dictionaries as well as objects and then we have a for loop that would convert these anything that isn't a dict with the `.to_dict()` method. As we'll want to remove these later and just use the object forms.

Reply:
Good point, I'll do that in the next PR